src/GeoSphere.cpp

```diff
@@ -441,8 +445,31 @@ void GeoSphere::Render(Graphics::Renderer *renderer, const matrix4x4d &modelView
 
 	renderer->SetTransform(matrix4x4f(modelView));
 
-	for (int i = 0; i < NUM_PATCHES; i++) {
-		m_patches[i]->Render(renderer, campos, modelView, frustum);
+	if (Pi::config->Int("SortGeoPatches") == 0) {
+		for (int i = 0; i < NUM_PATCHES; i++) {
+			m_patches[i]->Render(renderer, campos, modelView, frustum);
+		}
+	} else {
+		// Gather the patches that could be rendered
+		for (int i = 0; i < NUM_PATCHES; i++) {
+			m_patches[i]->GatherRenderablePatches(m_visiblePatches, renderer, campos, frustum);
+		}
+
+		// distance sort the patches
+		std::sort(m_visiblePatches.begin(), m_visiblePatches.end(), [&, campos](const GeoPatch *a, const GeoPatch *b) {
+			return (a->Centroid() - campos).LengthSqr() < (b->Centroid() - campos).LengthSqr();
+		});
```

Comment:
Additional note, storing the sort key (because that's what the camera distance is) in the vector rather than computing it from scratch with each iteration is significantly more cache-efficient as you're not pointer-chasing on every sort step.